pkg/prometheus/promcfg.go

```diff
@@ -2011,7 +2045,13 @@ func (cg *ConfigGenerator) generateAdditionalScrapeConfigs(
 				relabelings = append(relabelings, relabeling)
 			}
 		}
-		relabelings = cg.generateAddressShardingRelabelingRulesIfMissing(relabelings, shards)
+		// DaemonSet mode doesn't support sharding.
+		if !cg.daemonSet {
+			if shards == 1 {
+				relabelings = cg.generateAddressShardingRelabelingRulesIfMissing(relabelings, shards)
+			}
```

Comment:
this is why the unit tests fail IIUC
```suggestion
			relabelings = cg.generateAddressShardingRelabelingRulesIfMissing(relabelings, shards)
```